lib/index.js

```diff
@@ -345,13 +345,20 @@ exports.deepEqual = function (obj, ref, seen) {
 
 // Remove duplicate items from array
 
-exports.unique = function (array, key) {
+exports.unique = function (array, keys) {
+
+    if (keys && !Array.isArray(keys)) {
+        keys = [keys];
+    }
 
     var index = {};
     var result = [];
 
     for (var i = 0, il = array.length; i < il; ++i) {
-        var id = (key ? array[i][key] : array[i]);
+        var id = (keys ? keys.reduce(function(id, key) {
```

Comment:
Makes sense. I did a few more tests and the results are pretty much what I expect, the standard `for(i = 0...)` is the fastest and the difference between `Array.forEach()` and `Array.reduce()` is marginal.

The standard for loop is the clear winner here for single items, but for multiple keys they are significantly closer.

**Array with 750 Items**
```
  Hoek.uniqueReduce(array, key)  x 2,470 ops/sec ±2.69% (91 runs sampled)
  Hoek.uniqueLoop(array, key)    x 3,729 ops/sec ±0.93% (91 runs sampled)
  Hoek.uniqueForEach(array, key) x 2,514 ops/sec ±0.97% (92 runs sampled)

  Hoek.uniqueReduce(array, [keys])  x 1,041 ops/sec ±0.99% (93 runs sampled)
  Hoek.uniqueLoop(array, [keys])    x 1,301 ops/sec ±1.24% (92 runs sampled)
  Hoek.uniqueForEach(array, [keys]) x   938 ops/sec ±2.13% (80 runs sampled)
```

In this the for loop is clearly better, both `Array.reduce()` and `Array.forEach()` are about the same.

**Array with 4500 Items**
```
  Hoek.uniqueReduce(array, key)  x 396 ops/sec ±2.82% (83 runs sampled)
  Hoek.uniqueLoop(array, key)    x 629 ops/sec ±0.85% (92 runs sampled)
  Hoek.uniqueForEach(array, key) x 383 ops/sec ±3.20% (85 runs sampled)

  Hoek.uniqueReduce(array, [keys])  x 174 ops/sec ±1.11% (83 runs sampled)
  Hoek.uniqueLoop(array, [keys])    x 212 ops/sec ±1.61% (85 runs sampled)
  Hoek.uniqueForEach(array, [keys]) x 160 ops/sec ±1.16% (77 runs sampled)
```

Continuing the same trend here. I also did some tests with smaller arrays and as expected, the same trend.